hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -228,6 +232,9 @@ private void handlePlatformTransaction(
             // Dispatch the transaction to the handler
             dispatcher.dispatchHandle(context);
 
+            // Handle the end of a staking period
+            stakingPeriodTimeHook.process(consensusNow, context);
+
```

Comment:
This should happen **before** handling the first transaction. So, should it be before `dispatcher.dispatchHandle(context);` ?

Reply:
Ah, I didn't realize this happened _before_ the first transaction. I'll move it

Reply:
Done